sycl/test/regression/macro_conflict.cpp

```diff
@@ -0,0 +1,18 @@
+// RUN: %clangxx -fsyntax-only -Xclang -verify %s -o %t.out
+// expected-no-diagnostics
+//
+//===----------------------------------------------------------------------===//
+// This test checks if the user-defined macros SUCCESS, FAIL, BLOCKED are
+// defined in global namespace by sycl.hpp
+//===----------------------------------------------------------------------===//
+
+#define SUCCESS 0
+#define FAIL 1
+#define BLOCKED 2
+
+#include <CL/sycl.hpp>
+
```

Comment:
The code below `#include <CL/sycl.hpp>` is not relevant to the tested functionality and can be removed.

Reply:
I believe the compiler will optimize away the whole .cpp file if there is no main() function.
So, better leave it as is.

Reply:
Why do you care about optimization?
Maybe I'm missing something, but I think clang emits errors while parsing SYCL headers. Am I wrong?

Reply:
I think you are talking about linker error, right?
If so, please, add `-c` option - this should fix the problem.

Reply:
Yes, if main() is missing all the sycl.hpp code with wrong macro expansion will be optimized away which could hide the conflict issue.
Any particular reason why you want to make a C++ code without a one-line main()?

Reply:
> Yes, if main() is missing all the sycl.hpp code with wrong macro expansion will be optimized away which could hide the conflict issue.

Sorry, but I don't buy this. Optimization are applied after semantic checks, which are supposed to detect the conflict.
`main` function doesn't use SYCL code, so valid code will be optimized even with `main` function.

> Any particular reason why you want to make a C++ code without a one-line main()?

Yes - reproducers must be as small as possible to make it easy to read and fast to execute.

Reply:
Okay. Let me clarify the purpose of this test program.
This test program is intended to test if the user defined macro such as "#define SUCCESS 0" should not conflict with sycl.hpp and also should run successfully without any error.
So, I will change the main() function "return SUCCESS;" to make sure the user-defined macro is expanded successfully without conflicting with sycl. 
The test should run successfully with return value of zero.
Would you agree with this change?

Reply:
> and also should run successfully without any error.

1. I don't think that test should check for that.
2. It doesn't check that, it verifies only compilation step.

> So, I will change the main() function "return SUCCESS;" to make sure the user-defined macro is expanded successfully without conflicting with sycl.

The conflict is detected at compile time, so this check doesn't make sense to me.

Reply:
Because I fixed the conflict in SYCL compiler, there will be no conflict with the macro SUCCESS. :)
Again, the test is intended to RUN successfully, but not only checking syntax error.
The only reason why the user defined the macro is to use it in his program successfully.
We should check if the macro expansion worked fine by really RUNNING the test program.

Reply:
I disagree, but if you want to add execution test, please, add defines to one of the existing SYCL LIT tests, which enqueue something into the queue.

Reply:
Could you tell me how to add defines to one of the existing SYCL LIT tests?
Or if there is any wiki page that explains how to do it, I can take a look.

Reply:
I figured out that my regression test is already a LIT test. Sorry for ignorance.
I think combining my my test into an existing test file is not a good idea.
To isolate the purpose, it is better to keep the test case separate.
So, I removed the main() function as you requested.